lib/string-checker.js

```diff
@@ -77,38 +79,50 @@ StringChecker.prototype = {
 
     /**
      * Checks file provided with a string.
-     * @param {String} str
+     * @param {String} source
      * @param {String} [filename]
      * @returns {Errors}
      */
-    checkString: function(str, filename) {
+    checkString: function(source, filename) {
         filename = filename || 'input';
 
-        var tree;
+        var sourceTree;
         var parseError;
 
         try {
-            tree = JsFile.parse(str, this._esprima, this._configuration.getEsprimaOptions());
+            sourceTree = JsFile.parse(source, this._esprima, this._configuration.getEsprimaOptions());
         } catch (e) {
             parseError = e;
         }
 
-        var file = new JsFile(filename, str, tree, {
-            es3: this._configuration.isES3Enabled(),
-            es6: this._configuration.isESNextEnabled()
-        });
+        var file = this._createJsFileInstance(filename, source, sourceTree);
 
         var errors = new Errors(file, this._verbose);
-        var errorFilter = this._configuration.getErrorFilter();
 
         if (!this._maxErrorsExceeded) {
             if (parseError) {
-                errors.setCurrentRule('parseError');
-                errors.add(parseError.description, parseError.lineNumber, parseError.column);
-
+                this._addParseError(errors, parseError);
                 return errors;
             }
 
+            this._checkJsFile(file, errors);
+        }
+
+        return errors;
+    },
+
+    /**
+     * Checks a file specified using JsFile instance.
+     * Fills Errors instance with validation errors.
+     *
+     * @param {JsFile} file
+     * @param {Errors} errors
+     * @private
+     */
+    _checkJsFile: function(file, errors) {
+        var errorFilter = this._configuration.getErrorFilter();
+
+        if (!this._maxErrorsExceeded) {
```

Comment:
perhaps make this an early return now that it's an `if` enclosing almost the entire body of the function?